lib/src/push_notifications/src/push_notification_events.dart

```diff
@@ -4,7 +4,7 @@ import 'package:ably_flutter/ably_flutter.dart';
 
 /// Type definition for a callback invoked when background push message
 /// is received.
-typedef BackgroundMessageHandler = void Function(RemoteMessage message);
+typedef BackgroundMessageHandler = Future<void> Function(RemoteMessage message);
```

Comment:
It seems this change introduces breaking change, can we avoid this? We are not ready right now to create next major release of Flutter SDK

Reply:
I think this was needed to make the async await for it.  I'll play around with it.  I think we can change the type to be "dynamic" instead of void/Future<void>.  That might satisfy both situations.  

Is the issue it creates this one? 

```dart
 info • `Future` results in `async` function bodies must be `await`ed or marked `unawaited` using `dart:async` • lib/src/platform/src/method_call_handler.dart:97:5 • unawaited_futures
```

Reply:
Awesome! Thank you